allennlp/models/model.py

```diff
@@ -333,6 +335,61 @@ def extend_embedder_vocab(self, extended_vocab: Vocabulary) -> None:
             if isinstance(module, (Embedding, TextFieldEmbedder)):
                 module.extend_vocab(extended_vocab)
 
+    def inspection_dict(self,
+                        inspect: str = "parameters",
+                        quite: bool = False) -> OrderedDict:
+        """
+        Returns ordered nested dict showing the module / parameters paths and (optionally)
+        corresponding module names of the model. This can be helpful to setup model path
+        based regex, for example in initializer. If quite is false, dict will be printed.
+        Both outputs are less verbose than print(model).
+
+        Parameters
+        ----------
+        inspect: str, optional, (default = "parameters")
+            Inspects "parameters" or "modules" of the model.
+        quite: bool, optional, (default = False)
+            If true, doesn't print the output, only returns the final dict.
+        """
+        if inspect not in ["parameters", "modules"]:
+            ConfigurationError("model inspection works with modules or parameters only")
+
+        items = self.named_modules() if inspect == "modules" else self.named_parameters()
+        module_dict = {model_path: module for model_path, module in items}
+        sorted_module_paths = sorted(module_dict.keys(), key=lambda path: path.count('.'))
+
+        def nested_set(dict_, keys, value):
+            """
+            Given a nested list of keys defining a path, set's the value.
+            Eg. nested_set(dict_={}, ["one", "two"], "three") ==> {"one": {"two": "three"}}
+            """
+            for key in keys[:-1]:
+                dict_ = dict_.setdefault(key, {})
+            dict_[keys[-1]] = value
+
+        model_inspection_dict: OrderedDict[str, Any] = OrderedDict()
+        for module_path in sorted_module_paths:
+            module = module_dict[module_path]
+            module_class_name = module.__class__.__name__
+            if not module_path:
+                model_inspection_dict["__class__"] = module_class_name
+                continue
+            path_module_names = module_path.split(".")
+            value = {'__class__': module_class_name} if inspect == "modules" else "Parameter"
+            if path_module_names:
+                nested_set(model_inspection_dict, path_module_names, value)
```

Comment:
the operation of this whole code block is extremely opaque to me, I don't really get what it's trying to do. (it's producing a dict of some format, but what the format is and why I don't get)